MoltenVK/MoltenVK/GPUObjects/MVKDeviceMemory.h

```diff
@@ -136,41 +128,61 @@ class MVKDeviceMemory : public MVKVulkanAPIDeviceObject {
 					const VkMemoryAllocateInfo* pAllocateInfo,
 					const VkAllocationCallbacks* pAllocator);
 
-    ~MVKDeviceMemory() override;
+	~MVKDeviceMemory() override;
 
 protected:
 	friend class MVKBuffer;
-    friend class MVKImageMemoryBinding;
-    friend class MVKImagePlane;
+	friend class MVKImageMemoryBinding;
+	friend class MVKImagePlane;
 
 	void propagateDebugName() override;
 	VkDeviceSize adjustMemorySize(VkDeviceSize size, VkDeviceSize offset);
-	VkResult addBuffer(MVKBuffer* mvkBuff);
-	void removeBuffer(MVKBuffer* mvkBuff);
-	VkResult addImageMemoryBinding(MVKImageMemoryBinding* mvkImg);
-	void removeImageMemoryBinding(MVKImageMemoryBinding* mvkImg);
-	bool ensureMTLHeap();
-	bool ensureMTLBuffer();
-	bool ensureHostMemory();
-	void freeHostMemory();
-	MVKResource* getDedicatedResource();
-	void initExternalMemory(VkExternalMemoryHandleTypeFlags handleTypes);
-
-	MVKSmallVector<MVKBuffer*, 4> _buffers;
-	MVKSmallVector<MVKImageMemoryBinding*, 4> _imageMemoryBindings;
-	std::mutex _rezLock;
```

Comment:
I've removed this since my understanding is they are used to read/write to the resources that use this memory device objects. Allocated resources now will be placed in heaps and a buffer that spans the whole heap is provided to accomplish this. The only exception are imported resources, these are handled slightly different. The explanation is in code comments, but in short, textures require to be dedicated for export/import by VK_EXT_metal_objects, so we can create buffers when importing to account for host mapping due to Metal not providing anything; and buffers will just not have a heap if they are not backed by it at import.